drivers/peci/peci_ite_it8xxx2.c

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (c) 2022 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_peci_it8xxx2
+
+#include <drivers/gpio.h>
+#include <drivers/pinmux.h>
+#include <drivers/peci.h>
+#include <kernel.h>
+#include <errno.h>
+#include <device.h>
+#include <drivers/peci.h>
+#include <soc.h>
+#include <soc_dt.h>
+#include <logging/log.h>
+#include <sys/util.h>
+
+LOG_MODULE_REGISTER(peci_ite_it8xxx2, CONFIG_PECI_LOG_LEVEL);
+
+/* The following constants decribes the bitrate of it8xxx2 PECI,
+ * for the frequency are 2MHz, 1MHz, and 1.6MHz. (Unit: Hz)
+ */
+#define	PECI_IT8XXX2_BITRATE_2MHZ		2000000
+#define	PECI_IT8XXX2_BITRATE_1MHZ		1000000
+#define	PECI_IT8XXX2_BITRATE_1P6MHZ		1600000
+
+/* The following masks are designed for the PECI bitrate settings,
+ * for the bits[7:3] are not related to this features.
+ */
+#define	PECI_IT8XXX2_BITRATE_BITS_MASK		0x07
+#define	PECI_IT8XXX2_BITRATE_2MHZ_BITS		0x00
+#define	PECI_IT8XXX2_BITRATE_1MHZ_BITS		0x01
+#define	PECI_IT8XXX2_BITRATE_1P6MHZ_BITS	0x04
+
+/* The Transaction Timeout */
+#define PECI_TIMEOUT_MS		30
+
+/* PECI interface 0 */
+#define PECI0			0
+
+/* HOSTAR (F02C00h) */
+#define HOBY			BIT(0)
+#define FINISH			BIT(1)
+#define RD_FCS_ERR		BIT(2)
+#define WR_FCS_ERR		BIT(3)
+#define EXTERR			BIT(5)
+#define BUS_ER			BIT(6)
+#define TEMPERR			BIT(7)
+#define HOSTAR_RST_ANYBIT \
+		(TEMPERR+BUS_ER+EXTERR+WR_FCS_ERR+RD_FCS_ERR+FINISH)
+
+/* HOCTLR (F02C01h) */
+#define START			BIT(0)
+#define AWFCS_EN		BIT(1)
+#define CONTROL			BIT(2)
+#define PECIHEN			BIT(3)
+#define FCSERR_ABT		BIT(4)
+#define FIFOCLR			BIT(5)
+
+/*
+ * TODO: The Voltage Configuration
+ *       Related DTSi and registers settings should be fulfilled
+ *       in the future.
+ */
+/* PADCTLR (F02C0Eh) */
+#define PECI_DVIE		0x04
+
+enum peci_vtts {
+	HOVTTS0P85V = 0x00,
+	HOVTTS0P90V = 0x01,
+	HOVTTS0P95V = 0x02,
+	HOVTTS1P00V = 0x03,
+	HOVTTS1P05V = 0x08,
+	HOVTTS1P10V = 0x09,
+	HOVTTS1P15V = 0x0A,
+	HOVTTS1P20V = 0x0B,
+	HOVTTS1P25V = 0x10,
+};
+
+/* The following definitions are inclusive of the config/data
+ * and related properties of the PECI device.
+ */
+struct peci_alts_cfg {
+	const struct device *pinctrls;
+	uint8_t pin;
+	uint8_t alt_fun;
+};
+
+struct peci_it8xxx2_config {
+	uintptr_t base_addr;
+	uint8_t irq_no;
+	const struct peci_alts_cfg *alts_list;
+};
+
+struct peci_it8xxx2_data {
+	struct peci_msg *msgs;
+	struct k_sem device_sync_sem;
+	uint32_t bitrate;
+};
+
+static const struct peci_alts_cfg
+	peci_alt_inst[DT_INST_NUM_PINCTRLS_BY_IDX(0, 0)] =
+	IT8XXX2_DT_ALT_ITEMS_LIST(0);
+
+static const struct peci_it8xxx2_config peci_it8xxx2_config0 = {
+	.base_addr = DT_INST_REG_ADDR(0),
+	.irq_no = DT_INST_IRQN(0),
+	.alts_list = peci_alt_inst,
+};
+
+static struct peci_it8xxx2_data peci_it8xxx2_data0;
+
+/* ITE IT8XXX2 PECI Functions */
+
+static void peci_it8xxx2_set_dvie(const struct device *dev, bool en)
```

Comment:
Dear Keith, 

Thanks for guidance.
The function `peci_it8xxx2_set_dvie` has been removed because it's true that it should be set once in the initialization. 
Secondly, when it comes to passing a pointer to peci_regs instead of passing `dev`, I've checked some private functions and made some modifications. Please advise if there's any further suggestion.

Thanks again.
BJ